src/stories/Overlay.stories.tsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
```

Comment:
I pulled down the branch to check out the stories. I'm seeing a couple weird behaviors.

1. If I interact with one of the stories, then move to the other, the dialog doesn't appear when clicking the button.
2. If I go to the dialog story, open the overlay, close it, then open it again, the hover effects on the buttons are gone, and clicking anywhere on the overlay makes it disappear.

Reply:
I think this is a Portal bug 😭 I'm not sure why, but removing the `Portal` in `Overlay` seems to fix it. When `Overlay` does render wrapped with a `Portal` if you click around on one story, then move to the other, you can see in dev tools that the portal root doesn't get appended to the end of `BaseStyles` like it does in the first story 🤔

Reply:
Ooh, interesting. I will look into it tomorrow.

Reply:
@emplums This is resolved. I opened a PR targeting this branch to merge main and resolve. #1132

Reply:
@T-Hugs awesomeeee!! thank you!